### ISSUE TITLE
Parallax not working, as directed

### ISSUE THREAD

**Issue by akihitoalextsuboi** [Dogfalo/materialize#210]
Other js components working.
When appended class='parallax', image will disappear.
My slim file is like this.

``` top.html.slim
.parallax-container
  .parallax
    = image_tag 'top_hawaii.jpg', width: '100%'
```

**Comment by cchandurkar** [comment]
I am using 0.95.3 from cdnjs.cloudflare.com. It's still not working for me.

**Comment by allthesignals** [comment]
I'm still seeing this issue. I have materialize-sass 0.96.1. It works on first load, but navigating to new pages and returning to the page with the parallax effect disappears.

**Comment by iesen** [comment]
i am seeing this issue in 0.96.1. Image is hidden and not displayed..

**Comment by nosajnosliw** [comment]
I'm also seeing this and have been I was using .95 and now am using .96.1. any clues why this is happening? I need to launch my site today.

**Comment by allthesignals** [comment]
My issue was actually related to [turbolinks](http://stackoverflow.com/questions/18770517/rails-4-how-to-use-document-ready-with-turbo-links). If you're using Rails, double-check this.

**Comment by andersdn11** [comment]
I also can't get the parallax to work. 

Using CDN 0.97.0 

Any solutions to this?

**Comment by DmitrySharikov** [comment]
Same problem with 0.97

**Comment by DmitrySharikov** [comment]
I think I found solution for this problem.
@andersdn11 just delete any size settings for image

**Comment by andersdn11** [comment]
I dont have any size settings, still won't work. Could it be because i'm working localhost?

**Comment by DmitrySharikov** [comment]
I don't think so. I'm also working in localhost






__Best regards



Dmitry SharikovRuby on Rails developer





http://dmitrysharikov.github.io

**Comment by DavidCWebs** [comment]
I had this problem with v0.96.0 - fixed it by adding
~~~
(function($){
  $(function(){
    $('.parallax').parallax();
  });
})(jQuery);
~~~
...in the head rather than in the footer. Images are now visible, parallax working as expected.

**Comment by nosajnosliw** [comment]
I swapped out out the .95 js code for dropdowns into my .96 js and all works fine now.  Hope the new build supports parallax and dropdowns.

**Comment by andersdn11** [comment]
so i got it working. My problem seemed to be the z-index: -1 on parallax. class

Fix:
.parallax{ z-index: 0; }

**Comment by Sajgoniarz** [comment]
Same here. Not working in 0.97, tried in angular controller, script at page, but works only after firing it from console ;c

**Comment by DanAncona** [comment]
I'm getting this problem even with the parallax demo template on the getting started page. The web based demo runs and looks great, but when I download the zip and open index.html locally, the images aren't showing up for me. Chrome and Safari/OS X. It almost works if I set the z-index to 0 - the button stays visible but the head and subhead disappear and even setting a higher z-index on them doesn't make them reappear. Weird stuff.

**Comment by HarryMcKinney** [comment]
No images at first, in the footer was this: <script src="../../dist/js/materialize.js"></script> and I changed it to: <script src="js/materialize.js"></script> and now it works.

**Comment by gravityshouldbenaut** [comment]
Yeah I checked my footer and saw that the script went to bin/materialize.js. Just changed it to js/materialize.js and everything was fine afterwards

**Comment by osleg** [comment]
version 0.97.6 - parallax not working, div class="parallax" hiding the image

**Comment by lrjbrual** [comment]
Help it is not working 

  .parallax img{
   background:url("background1.jpg") no-repeat 0% 0%;
   background-size:cover;
}

**Comment by marcemc** [comment]
for some reason css is:
```
 .parallax img{
      display:none;
      ...
}
```
overriding solved the problem:
 ```
.parallax img{
      display: inherit !important;
}
```

**Comment by jtkDvlp** [comment]
Same problem:

```
        $this.children("img").one("load", function() {
          updateParallax(true);
        }).each(function() {
          if(this.complete) $(this).load(); // seems to be the problem-function
        });
```

> jQuery.Deferred exception: Cannot read property 'indexOf' of undefined TypeError: Cannot read property 'indexOf' of undefined
>     at r.fn.load (http://localhost/projekte/homepage/extern/jquery/dist/jquery.min.js:4:18685)
>     at HTMLImageElement.<anonymous> (http://localhost/projekte/homepage/extern/Materialize/js/parallax.js:42:37)
>     at Function.r.extend.each (http://localhost/projekte/homepage/extern/jquery/dist/jquery.min.js:2:2813)
>     at r.fn.r.each (http://localhost/projekte/homepage/extern/jquery/dist/jquery.min.js:2:1003)
>     at HTMLDivElement.<anonymous> (http://localhost/projekte/homepage/extern/Materialize/js/parallax.js:41:12)
>     at Function.r.extend.each (http://localhost/projekte/homepage/extern/jquery/dist/jquery.min.js:2:2813)
>     at r.fn.r.each (http://localhost/projekte/homepage/extern/jquery/dist/jquery.min.js:2:1003)
>     at $.fn.parallax (http://localhost/projekte/homepage/extern/Materialize/js/parallax.js:6:19)
>     at HTMLDocument.<anonymous> (http://localhost/projekte/homepage/init.js:2:18)
>     at j (http://localhost/projekte/homepage/extern/jquery/dist/jquery.min.js:2:29588) undefined

**Comment by SCLeoX** [comment]
Seriously, for me, using jQuery 2.2.4 solved the problem. jQuery 3 has a lot of compatible problems now. >.>

**Comment by m45rud** [comment]
i think it come from version of jquery. Try to user jquery from demo website and it works fine for me.

**Comment by iyedg** [comment]
The problem is definitely compatibility with jQuery 3, downgrading to 2.2.4 solved the issue for me too.

**Comment by JGeraldoLima** [comment]
I'm trying to use the Parallax Template, but only the first image is not showing. When I open the browser console developer, a part of it is shown. I've tried everything explained in this issue and on https://krescruz.github.io/angular-materialize/ , but I'm not getting success.

Plus, I'm using  an yoman angular-generator project. Any ideia?

**Comment by SCLeoX** [comment]
@JGeraldoLima What version of jQuery are you using? If not, please try 2.2.4. :)

**Comment by Raiper34** [comment]
Version 2.2.4 works for me, thanks

**Comment by JGeraldoLima** [comment]
hey @SCLeoX , sorry for the delay. I made it by editing a few .css properties. I don't know why it not worked automatically, but that's enough for now! Thank you very much.

**Comment by thebiltheory** [comment]
jQuery 2.2.4 fixed the issue and deleted the console indexOf Error.

**Comment by NitroBAY** [comment]
Damn I REALLY would like to use 3.1
any idea @Dogfalo or any other of the team ? Because #3570 should have solve that.

**Comment by olvrcrls** [comment]
on the .css file edit the display property
`.parallax img {
   display: block;
}`

**Comment by IFexe** [comment]
If it is necessary to work exactly in the block that is needed, then replace $(window) with $('.you-class')

<div class="section-slider">
				<div id="section-slide-1" class="section-slider__item active">
					<div class="list">
						<div class="list__header parallax-container">
							<div class="parallax">
								<img src="img/parallax.png" alt="">
							</div>
							<div class="list__header__text">
								
							</div>
						</div>
						<div class="list__body">
							
						</div>
					</div>
				</div>
				<div id="section-slide-2" class="section-slider__item">2</div>
				<div id="section-slide-3" class="section-slider__item">3</div>
				<div id="section-slide-4" class="section-slider__item">4</div>
				<div id="section-slide-5" class="section-slider__item">5</div>
				<div id="section-slide-6" class="section-slider__item">6</div>
			</div>


(function ($) {

  $.fn.parallax = function () {
    var window_width = $('.you-class').width();
    // Parallax Scripts
    return this.each(function(i) {
      var $this = $(this);
      $this.addClass('parallax');

      function updateParallax(initial) {
        var container_height;
        if (window_width < 601) {
          container_height = ($this.height() > 0) ? $this.height() : $this.children("img").height();
        }
        else {
          container_height = ($this.height() > 0) ? $this.height() : 500;
        }
        var $img = $this.children("img").first();
        var img_height = $img.height();
        var parallax_dist = img_height - container_height;
        var bottom = $this.offset().top + container_height;
        var top = $this.offset().top;
        var scrollTop = $('.you-class').scrollTop();
        var windowHeight = window.innerHeight;
        var windowBottom = scrollTop + windowHeight;
        var percentScrolled = (windowBottom - top) / (container_height + windowHeight);
        var parallax = Math.round((parallax_dist * percentScrolled));

        if (initial) {
          $img.css('display', 'block');
        }
        if ((bottom > scrollTop) && (top < (scrollTop + windowHeight))) {
          $img.css('transform', "translate3D(-50%," + parallax + "px, 0)");
        }

      }

      // Wait for image load
      $this.children("img").one("load", function() {
        updateParallax(true);
      }).each(function() {
        if (this.complete) $(this).trigger("load");
      });

      $('.you-class').scroll(function() {
        window_width = $('.you-class').width();
        updateParallax(false);
      });

      $('.you-class').resize(function() {
        window_width = $('.you-class').width();
        updateParallax(false);
      });

    });

  };

**Comment by tomscholz** [comment]
@NitroBAY update to Jquery 3 will be shipped with the next release

**Comment by afzalakhtar** [comment]
Parallax working fine on Mobile and Desktop view but not working on ipad view only

**Comment by vivekascoder** [comment]
I am having errors after using JQuery 2.2.4.
The Error Showing:

materialize.js:3890 Uncaught TypeError: Cannot read property 'offsetHeight' of undefined
    at Parallax._updateParallax (materialize.js:3890)
    at new Parallax (materialize.js:3830)
    at Function.init (materialize.js:1044)
    at Function.init (materialize.js:3909)
    at jQuery.fn.init.jQuery.fn.<computed> [as parallax] (materialize.js:1148)
    at HTMLDocument.<anonymous> (index.html:331)
    at fire (jquery.js:3187)
    at Object.fireWith [as resolveWith] (jquery.js:3317)
    at Function.ready (jquery.js:3536)
    at HTMLDocument.completed (jquery.js:3552)

**Comment by tundachef** [comment]
`  <div class="parallax"><img src="images/cow.jpg" style="transform: translate3d(-50%, 432.451px, 0px);opacity: 1;"></div>`
Adding this made the image appear...

**Comment by Alextastico** [comment]
> ` <div class="parallax"><img src="images/cow.jpg" style="transform: translate3d(-50%, 432.451px, 0px);opacity: 1;"></div>`
> Adding this made the image appear...

Not Working :(

**Comment by 96312** [comment]
@Alextastico Try adding just the `style="opacity: 1;"` inside your image tag. It works without this on local host but not on deployment.
